ush/python/pygfs/task/upp.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+
+import os
+from logging import getLogger
+from typing import Dict, Any, Union
+from pprint import pformat
+
+from pygw.attrdict import AttrDict
+from pygw.yaml_file import parse_j2yaml
+from pygw.file_utils import FileHandler
+from pygw.jinja import Jinja
+from pygw.logger import logit
+from pygw.task import Task
+from pygw.timetools import add_to_datetime, to_timedelta
+from pygw.exceptions import WorkflowException
+from pygw.executable import Executable, which
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class UPP(Task):
+    """Unified Post Processor Task
+    """
+
+    VALID_UPP_RUN = ['analysis', 'forecast', 'goes']
+
+    @logit(logger, name="UPP")
+    def __init__(self, config: Dict[str, Any]) -> None:
+        """Constructor for the UPP task
+        The constructor is responsible for resolving the "UPP_CONFIG" based in the run-type "upp_run"
+        Valid options for "upp_run" are:
+        analysis: process analysis output
+        forecast: process UFS-weather-model forecast output
+        goes: process UFS-weather-model forecast output for simulated satellite imagery
+
+        Parameters
+        ----------
+        config : Dict[str, Any]
+            Incoming configuration for the task from the environment
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(config)
+
+        valid_datetime = add_to_datetime(self.runtime_config.current_cycle, to_timedelta(f"{self.config.FORECAST_HOUR}H"))
+
+        localdict = AttrDict(
+            {'upp_run': self.config.UPP_RUN,
+             'forecast_hour': self.config.FORECAST_HOUR,
+             'valid_datetime': valid_datetime,
+             'atmos_filename': f"atm_{valid_datetime.strftime('%Y%m%d%H%M%S')}.nc",
+             'flux_filename': f"sfc_{valid_datetime.strftime('%Y%m%d%H%M%S')}.nc"
+             }
+        )
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **localdict)
+
+        if localdict.upp_run not in UPP.VALID_UPP_RUN:
```

Comment:
Rather than maintaining a valid list in the python file, can we just go based on the contents of the yaml file? Iff the `upp_run` does not match an entry in `upp.yaml`, it is invalid.

Reply:
That seems like a feature, not a bug. Is there a reason just adding a new upp_run block with all the necessary settings in the yaml wouldn't work or would be undesirable?

Reply:
Sure.  In that case, there is no need to check a valid run. Anything is valid.